packages/wdio-allure-reporter/src/index.ts

```diff
@@ -1,605 +1,686 @@
-     * @param {string} value - argument value
-     */
-    static addArgument = (name, value) => {
-        tellReporter(events.addArgument, { name, value })
-    }
-}
-
-export default AllureReporter
+import WDIOReporter, { SuiteStats, Tag, HookStats, TestStats, BeforeCommandArgs, AfterCommandArgs, CommandArgs, WDIOReporterBaseOptions } from '@wdio/reporter'
+import Allure from 'allure-js-commons'
+import Step from 'allure-js-commons/beans/step'
+import { getTestStatus, isEmpty, tellReporter, isMochaEachHooks, getErrorFromFailedTest, isMochaAllHooks, getLinkByTemplate } from './utils'
+import { events, PASSED, PENDING, SKIPPED, stepStatuses } from './constants'
+
+/**
+ * When you add a new option, please also update the docs at ./packages/wdio-allure-reporter/README.md
+ */
+interface AllureReporterOptions extends WDIOReporterBaseOptions {
+    /**
+     * defaults to `./allure-results`. After a test run is complete, you will find that this directory
+     * has been populated with an `.xml` file for each spec, plus a number of `.txt` and `.png`
+     * files and other attachments.
+     */
+    outputDir?: string;
+    /**
+     * optional parameter (`false` by default), set it to true in order to change the report hierarchy
+     * when using cucumber. Try it for yourself and see how it looks.
+     */
+    useCucumberStepReporter?: boolean;
+    /**
+     * optional parameter (`false` by default), set it to true in order to not fetch the `before/after`
+     * stacktrace/screenshot/result hooks into the Allure Reporter.
+     */
+    disableMochaHooks?: boolean;
+    /**
+     * optional parameter(`false` by default), in order to log only custom steps to the reporter.
+     */
+    disableWebdriverStepsReporting?: boolean;
+    /**
+     * optional parameter(`false` by default), in order to not attach screenshots to the reporter.
+     */
+    disableWebdriverScreenshotsReporting?: boolean;
+    /**
+     * optional parameter, in order to specify the issue link pattern. Reporter will replace `{}` placeholder
+     * with value specified in `addIssue(value)` call parameter.
+     * Example `https://example.org/issue/{}`
+     */
+    issueLinkTemplate?: string;
+    /**
+     * optional parameter, in order to specify the tms link pattern. Reporter will replace `{}` placeholder
+     * with value specified in `addTestId(value)` call parameter. Example `https://example.org/tms/{}`
+     */
+    tmsLinkTemplate?: string;
+}
+
+class AllureReporter extends WDIOReporter {
+    allure: Allure;
+    capabilities: WebDriver.DesiredCapabilities;
+    config: any;
+    isMultiremote?: boolean;
+    lastScreenshot?: string;
+    options: AllureReporterOptions;
+
+    constructor(options: AllureReporterOptions = {}) {
+        const outputDir = options.outputDir || 'allure-results'
+        super({
+            ...options,
+            outputDir,
+        })
+        this.options = options
+        this.config = {}
+        this.capabilities = {}

```

Comment:
we can use default values when defining them in the header or get rid of this entirely.

```suggestion
```

Reply:
If we define `capabilities` as 
```js
private _capabilities: WebDriver.DesiredCapabilities;
```
we also need to update where we use it to `this._capabilities`

Reply:
This is not possible will will result in more ugly code because `super` must be the first call if we have default assignment outside of constructor, and we want `outputDir` to be `allure-results` by default